### PR TITLE
Issue/19 add redirect

### LINKED ISSUE #19: Download zip files from repos
For example you have `https://github.com/smirnovanv/node-basics.git` repo and link `https://github.com/smirnovanv/node-basics/archive/refs/heads/master.zip` (note that it can contain redirect by 301 or 302, so your query should follow redirects)

### PR DESCRIPTION
**Pull request by smirnovanv**
Closes #19

### CODE REVIEW

File: src/fetchFile.ts [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
   link: string, 
   folderName: string, 
   fsLike: typeof fs | any = fs, 
   httpsLike: typeof https | any = https,
 ) => {
+  
   return new Promise<void>((resolve, reject) => {
     const localFileName = getLocalFileName(link);
+    const fileNameSplit = localFileName.split('.');
+    const fileType = fileNameSplit[fileNameSplit.length - 1];
```
Comment (m0sk1t) [inline src/fetchFile.ts:19]: `slice(-1)` or better to use `path.parse()`

File: src/fetchFile.ts [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
     const file = fsLike.createWriteStream(`${folderName}/${localFileName}`);
-
     httpsLike.get(link, function(response) {
       response.pipe(file);
+      
       file.on('finish', function(){
+        if (fileType === 'zip') {
+          unzipFile(folderName, localFileName);
```
Comment (m0sk1t) [inline src/fetchFile.ts:26]: move unzip and hash creation actions outside fetch file

**Review by m0sk1t — CHANGES_REQUESTED** [review]

File: src/__tests__/getLocalFileName.spec.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import { getLocalFileName } from '../getLocalFileName';
+
+describe('test getting local file name', () => {
+  const url = 'https://filesamples.com/samples/document/txt/sample3.txt'
+  const { name, ext } = path.parse(url);
+  const result = getLocalFileName(url);
+  const hashLength = 96;
+
+  test('local file name starts with correct name', () => {
+    expect(result.slice(0, name.length + 1)).toBe('sample3-');
```
Comment (m0sk1t) [inline src/__tests__/getLocalFileName.spec.ts:12]: nit: String.[startsWith|endsWith] consider to try it

File: src/__tests__/getLocalFileName.spec.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  });
+
+  test('local file name has right extention', () => {
+    expect(result.slice(name.length + hashLength + 1)).toBe('.txt');
```
Comment (m0sk1t) [inline src/__tests__/getLocalFileName.spec.ts:16]: endsWith

File: src/index.ts [hunk 5]
```diff
@@ -25,16 +24,19 @@ const downloadFilesIntoTempFolder = async (downloadLinks?: string[]) => {
...
+  return tempFolderName;
 }
 
 const incomingLinks = process.argv.slice(2);
 
 if (incomingLinks.length > 0) {
-  downloadFilesIntoTempFolder(incomingLinks);
+  (async() => {
+    const tempFolder = await downloadFiles(incomingLinks);
+    showData(tempFolder);
```
Comment (m0sk1t) [inline src/index.ts:38]: + tests for showData

**Review by m0sk1t — CHANGES_REQUESTED** [review]
just small changes

File: src/fetchFile.ts (hunk 2 above)
Comment (smirnovanv) [reply src/fetchFile.ts]: done

File: src/__tests__/getLocalFileName.spec.ts (hunk 3 above)
Comment (smirnovanv) [reply src/__tests__/getLocalFileName.spec.ts]: done

File: src/__tests__/getLocalFileName.spec.ts (hunk 4 above)
Comment (smirnovanv) [reply src/__tests__/getLocalFileName.spec.ts]: done

File: src/index.ts (hunk 5 above)
Comment (smirnovanv) [reply src/index.ts:38]: done

**Review by m0sk1t — APPROVED** [review]

**Pull request merged**